design/gas/Design_gas.md

```diff
@@ -0,0 +1,72 @@
+
+
+### Initial Bootstrapping of Network Gas
+
+Similar to other layer 2 protocols, Obscuro seeks to leverage Ethereum for gas usage. To facilitate this, Ethereum must first be bridged over to Obscuro. Although the bridge typically operates with a relayer, relaying a message incurs a gas cost. Hence, to bootstrap the entire layer 2 protocol, including relayers and bridges, cross-chain messages concerning Ethereum depositing into an account will be automatically executed on layer 2 at no cost and without the need for relaying. This ensures no additional transactions that necessitate publishing to layer 1.
+    
+
+### Types of Transactions
+
+Ethereum has developed to accommodate several [transaction types](https://docs.infura.io/networks/ethereum/concepts/transaction-types):
+1) Regular transactions - they work based on a gas auction.
+2) Access list transactions - same as previous ones, but they also announce what storage access they are supposed to perform when executed.
+3) EIP-1559 - the new transactions that burn the gas fee and work with tips in order to achieve a fair system.
+
+Usually layer 2's support EIP-1559 so there is a pending question of what we would need to support. 
+
+
+## Specification
+
+1. Ethereum must be used for gas operations and bridged from the layer 1 protocol.
+2. Bridged gas should automatically deposit into the respective addresses.
+3. Gas-consuming transactions should include a layer 1 cost component for calldata publishing and a layer 2 component that covers execution costs and prevents endless transaction execution.
+4. L2 gas price should be configurable. Furthermore it should support being free whilst still not allowing non halting execution.
+
+## Gas Cost Display
+
+Metamask can disassemble the cost components for layer 1 and layer 2 for specific transactions on Optimism, such as token balance approvals. However, it remains unclear how they achieve this as it's neither fully documented nor supports all transactions.
+
+
+## Design for Gas Deposit
+
+Given that Ethereum as a currency operates on the layer 1 protocol, it needs to be bridged to Obscuro for usage. This transition mechanism will be facilitated by the standard Obscuro bridge. The deposited value will produce a cross-chain message indicating the Ethereum recipient through the smart contract.
+
+```EthereumDeposited(address receiver, uint256 amount)```
+
+The enclave will automatically pick up this message, along with other cross-chain messages, but it will be treated specially. Upon detection, the enclave will increase the balance of the receiver account by the amount specified in the message.
```

Comment:
No relayer required?

Reply:
No. If we require a relayer here, we end up with a chicken and an egg problem where during the initial bootstrap of the network no one has ETH to their address and thus no one can relay the message. This is the only special handling required of us in order to ensure the network can always receive ETH and work with it. Note that those messages will not end up in calling the address and potentially a fallback function if someone is depositing to a smart contract; It's just a stateDB balance increase.